crates/sui/src/unit_tests/cli_tests.rs

```diff
@@ -555,6 +556,147 @@ async fn test_package_publish_command() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_publish_command_with_unpublished_dependency_succeeds(
+) -> Result<(), anyhow::Error> {
+    let with_unpublished_dependencies = true; // Value under test, results in successful response.
+
+    let mut test_cluster = TestClusterBuilder::new().build().await?;
+    let address = test_cluster.get_address_0();
+    let context = &mut test_cluster.wallet;
+
+    let client = context.get_client().await?;
+    let object_refs = client
+        .read_api()
+        .get_objects_owned_by_address(address)
+        .await?;
+
+    // Check log output contains all object ids.
```

Comment:
Nit: I think this comment is stale in all the new tests.

Reply:
whoops, copy pasta :(